build/index.js

```diff
@@ -5,35 +5,68 @@
  * @author Andrew Sliwinski <andrew@diy.org>
  */
 
+/*globals Promise:true*/
```

Comment:
I'd rather not introduce promises to this. This should be simple enough to be broken down into smaller functions to avoid callback hell.